policy/modules/services/git.if

```diff
@@ -47,6 +47,67 @@ interface(`git_role',`
 	')
 ')
 
+########################################
+## <summary>
+##	Role access for Git client.
+## </summary>
+## <param name="role_prefix">
+##	<summary>
+##	The prefix of the user role (e.g., user
+##	is the prefix for user_r).
+##	</summary>
+## </param>
+## <param name="user_role">
+##	<summary>
+##	The role associated with the user domain.
+##	</summary>
+## </param>
+## <param name="user_domain">
+##	<summary>
+##	The type of the user domain.
+##	</summary>
+## </param>
+#
+template(`git_client_role_template',`
+	gen_require(`
+		attribute git_client_domain;
+		type git_exec_t, git_home_t;
+	')
+
+	########################################
+	#
+	# Declarations
+	#
+
+	type $1_git_t, git_client_domain;
+	userdom_user_application_domain($1_git_t, git_exec_t)
+	role $2 types $1_git_t;
+
+	########################################
+	#
+	# Policy
+	#
+
+	allow $3 git_home_t:dir { manage_dir_perms relabel_dir_perms };
+	allow $3 git_home_t:file { exec_file_perms manage_file_perms relabel_file_perms };
+	userdom_user_home_dir_filetrans($3, git_home_t, dir, ".git")
+
+	allow $3 $1_git_t:process { ptrace signal_perms };
+	ps_process_pattern($3, $1_git_t)
+
+	auth_use_nsswitch($1_git_t)
+
+	# allow userdomains to exec git hooks
+	exec_files_pattern($3, git_home_t, git_home_t)
+	# transition back to the user domain when executing git hooks
+	domtrans_pattern($1_git_t, git_home_t, $3)
```

Comment:
I understand the logic of this, but it allows the domain to execute arbitrary code. It could write a `git_home_t` file and then execute it. Since it runs in the user's domain, it could access other secrets.

If we keep the line, please put at the top of the rule block (line 90)

Reply:
I would like to keep this logic because it allows the git client to operate in a way that I as a user would expect. Do you think putting this behind a tunable makes more sense?

Reply:
Alternatively we could create a separate type for git hooks and only allow those to be executed, but not written to by the client.

Reply:
I think a type for hooks is a better option.

Reply:
I have added the `git_home_hook_t` type for git hooks. This type is pretty much identical to `git_home_t` except `git_client_domain` does not have write access to it (and we also remove git's access to execute `git_home_t`). Unfortunately, we can't easily allow the git client to create these files because they are copied from `/usr/share/git-core/templates`, and thus `write` access is required. Instead, a user will need to manually relabel git hooks to this type in order to allow git to execute them.